vm/systemSmartContracts/esdt_test.go

```diff
@@ -4625,4 +4631,146 @@ func registerAndSetAllRolesWithTypeCheck(t *testing.T, typeArgument []byte, expe
+	assert.True(t, strings.Contains(eei.returnMessage, "invalid number of arguments, wanted 1"))
+
+	owner := bytes.Repeat([]byte{1}, 32)
+	tokenName := []byte("TOKEN-ABABAB")
+	tokensMap := map[string][]byte{}
+	marshalizedData, _ := args.Marshalizer.Marshal(ESDTDataV2{
+		TokenName:    tokenName,
+		OwnerAddress: owner,
+		CanPause:     true,
+		IsPaused:     true,
+	})
+	tokensMap[string(tokenName)] = marshalizedData
+	eei.storageUpdate[string(eei.scAddress)] = tokensMap
+
+	vmInput.CallerAddr = owner
+	vmInput.Arguments = [][]byte{tokenName}
+	output = e.Execute(vmInput)
+	assert.Equal(t, vmcommon.Ok, output)
+
+	vmOutput := eei.CreateVMOutput()
+
+	systemAddress := make([]byte, len(core.SystemAccountAddress))
+	copy(systemAddress, core.SystemAccountAddress)
+	systemAddress[len(core.SystemAccountAddress)-1] = 0
+
+	createdAcc, accCreated := vmOutput.OutputAccounts[string(systemAddress)]
+	assert.True(t, accCreated)
+
+	assert.True(t, len(createdAcc.OutputTransfers) == 1)
+	outputTransfer := createdAcc.OutputTransfers[0]
+
+	assert.Equal(t, big.NewInt(0), outputTransfer.Value)
+	expectedInput := vmcommon.BuiltInFunctionESDTSetBurnRoleForAll + "@" + hex.EncodeToString(tokenName)
+	assert.Equal(t, []byte(expectedInput), outputTransfer.Data)
+	assert.Equal(t, vmData.DirectCall, outputTransfer.CallType)
+
+	output = e.Execute(vmInput)
+	assert.Equal(t, vmcommon.UserError, output)
+	assert.True(t, strings.Contains(eei.returnMessage, "cannot set burn for all as it was already set"))
+}
+
+func TestEsdt_unsetBurnForAll(t *testing.T) {
+	t.Parallel()
+
+	args := createMockArgumentsForESDT()
+	eei, _ := NewVMContext(
+		&mock.BlockChainHookStub{},
+		hooks.NewVMCryptoHook(),
+		&mock.ArgumentParserMock{},
+		&stateMock.AccountsStub{},
+		&mock.RaterMock{})
+	args.Eei = eei
+
+	e, _ := NewESDTSmartContract(args)
+	vmInput := getDefaultVmInputForFunc("unsetBurnForAll", [][]byte{})
+
+	e.flagBurnForAll.Reset()
+	output := e.Execute(vmInput)
+	assert.Equal(t, vmcommon.FunctionNotFound, output)
+	assert.True(t, strings.Contains(eei.returnMessage, "invalid method to call"))
+
+	e.flagBurnForAll.SetValue(true)
+	output = e.Execute(vmInput)
+	assert.Equal(t, vmcommon.FunctionWrongSignature, output)
+	assert.True(t, strings.Contains(eei.returnMessage, "invalid number of arguments, wanted 1"))
+
+	owner := bytes.Repeat([]byte{1}, 32)
+	tokenName := []byte("TOKEN-ABABAB")
+	tokensMap := map[string][]byte{}
+	token := &ESDTDataV2{
+		TokenName:    tokenName,
+		OwnerAddress: owner,
+		CanPause:     true,
+		IsPaused:     true,
+	}
+
+	burnForAllRole := &ESDTRoles{Roles: [][]byte{[]byte(vmcommon.ESDTRoleBurnForAll)}, Address: []byte{}}
+	token.SpecialRoles = append(token.SpecialRoles, burnForAllRole)
+
+	marshalizedData, _ := args.Marshalizer.Marshal(token)
+	tokensMap[string(tokenName)] = marshalizedData
+	eei.storageUpdate[string(eei.scAddress)] = tokensMap
+
+	vmInput.CallerAddr = owner
+	vmInput.Arguments = [][]byte{tokenName}
+	output = e.Execute(vmInput)
```

Comment:
linter: `ineffectual assignment to output (ineffassign)`

Reply:
done